### ISSUE TITLE
QC: fully document both "analytics" and "reporting" data models in a vignette

### ISSUE THREAD

**Issue by jwildfire** [Gilead-BioStats/gsm#1561]
## QC Details
- Analytics data model applies data generated using to KRI/Country workflows (See existing data model refactor issue)
   - `dfInput`
   - `dfTransofrmed`
   - `dfAnalyzed`
   - `dfBounds`
   - `dfFlagged`
   - `dfSummary`
- Reporting data model applies to data used by charting and reporting functions
  - `dfSummary`
  - `dfBounds`
  - `dfSite`
  - `dfStudy`
  - `dfMetrics`

Specify all columns along with type and whether it is optional or required. 

Fine to throw this in a discussion or wiki page for now, we can convert to vignette and/or data files later.

**Comment by jwildfire** [comment]
Once this is done, review `rbm-viz` requirements, update this requirement as needed, and file issues to synchronize.

**Comment by samussiah** [comment]
Might start with [`rbm-viz` data specs](https://github.com/Gilead-BioStats/rbm-viz/wiki/Data-Specifications) and update accordingly.

**Comment by jwildfire** [comment]
see https://github.com/Gilead-BioStats/gsm/issues/1538 for analytics model

**Comment by jwildfire** [comment]
for reporting we want a table comparing the current rbm-viz data model with the current gsm v2 model.

**Comment by lauramaxwell** [comment]
@jwildfire @samussiah got started on a wiki entry for this https://github.com/Gilead-BioStats/gsm/wiki/Data-Model-(%E2%9A%A0%EF%B8%8FWork-in-Progress%E2%9A%A0%EF%B8%8F)

only the `Analytics data model` section is flushed out, the `Reporting` section is just the shell copied over from `dfInput` at the moment.

**Comment by lauramaxwell** [comment]
@lauramaxwell to-do:  Add column for `rbmViz` column name in the reporting to be able to compare and determine if/how to reconcile differences

**Comment by jwildfire** [comment]
Wiki page looks great! We can add as an appendix in the Data Pipeline Vignette when we're ready.

**Comment by jwildfire** [comment]
I wonder if we should refactor `dfSites` to be `dfGroups` and include country metadata ... 

@samussiah @lauramaxwell thoughts ?

**Comment by samussiah** [comment]
I was considering the same.  Some metadata applies across group levels:

- group ID
- group label
- group status (kind of)
- # of participants

But most site-level attributes don't really apply to countries.

**Comment by jwildfire** [comment]
How about the following for a new `dfGroups` object in the reporting model: 

- ProtocolID
- GroupID
- GroupLevel
- Enrollment
- InvestigatorName*
- InvestigatorNumber* - Last, First Name
- Status* 
- City*
- State*
- Country*

* - Expected When GroupLevel = "Site"

**Comment by jwildfire** [comment]
Probably want a similar refactor for `dfStudy`. Then would want to create new workflows to deal with this ...

I'll try to file a separate issue and work on a POC tomorrow.

**Comment by lauramaxwell** [comment]
I like this, and I prefer the optional fields that are more explicit to the `Details` field, unless we need it to be able to contain an aribitrary number of group-specific fields. Either way, I'll update the wiki once we land on the column names.

Aside from examples and a couple references in the reporting functions, it looks like the edits to the codebase for the `dfStudy` and `dfSite` column names will primarily be in `rbm-viz`, right?
